Source/JavaScriptCore/runtime/JSArrayBufferView.cpp

```diff
@@ -324,6 +325,60 @@ RefPtr<ArrayBufferView> JSArrayBufferView::possiblySharedImpl()
     }
 }
 
+bool isIntegerIndexedObjectOutOfBounds(JSArrayBufferView* typedArray, IdempotentArrayBufferByteLengthGetter& getter)
+{
+    // https://tc39.es/proposal-resizablearraybuffer/#sec-isintegerindexedobjectoutofbounds
+
+    if (typedArray->isDetached())
+        return true;
+
+    if (!isResizable(typedArray->mode()))
+        return false;
```

Comment:
Why does not resizable automatically mean not out of bounds?

Oh, the opposite of "out of bounds" here is not "not out of bounds".  Instead, it is "we don't know".  I guess that makes sense.  Maybe add a comment that states this?  It's very easy to read returning false as declaring "not out of bounds".

Reply:
This is aligned to the spec precisely, see https://tc39.es/proposal-resizablearraybuffer/#sec-isintegerindexedobjectoutofbounds (above link in this code).
This is checking whether this is detached technically.